werkzeug/datastructures.py

```diff
@@ -62,7 +66,10 @@ def setmethod(cls, name):
 
     def wrap(cls):
         for name in names:
-            setmethod(cls, name)
+            if PY2:
```

Comment:
IMO the behavior should be exactly as with the builtin dict, that is, provide viewmethods only on Python 2.